animated_sprite.h

```diff
@@ -0,0 +1,94 @@
+#ifndef ANIMATEDSPRITE_H
+#define ANIMATEDSPRITE_H
+
+#include <SDL2/SDL.h>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "sprite.h"
+#include "vector2.h"
+
+class AnimatedSprite : public Sprite {
+ public:
+  // Constructor takes the name of the sprite sheet, the source rectangle for
+  // the first frame, the renderer, and the time between each frame.
+  AnimatedSprite(std::string sprite_sheet_name,
+                 SDL_Rect first_sprite_coord_location, SDL_Renderer* renderer,
+                 float time_to_update_sprite, SDL_Rect destination);
+
+  // Destructor.
+  virtual ~AnimatedSprite() = default;
+  // Checks to see if it is time to move to the next frame in the current
+  // animation.
+  void TimeToUpdateFrame();
+
+  // Gets current_animation.
+  std::string GetCurrentAnimation();
+
+  // Gets destination_rect_.
+  SDL_Rect GetDestinationRect();
+
+  // Gets frame_index_.
```

Comment:
nit: These comments are fine but a little technical. Consider writing them like the person reading doesn't know anything about the internals of the class.

```
// Returns the index to the current frame being displayed for this animated sprite.
int GetFrameIndex();
```